lib/scanner/walk.go

```diff
@@ -402,17 +402,31 @@ func (w *walker) walkRegular(ctx context.Context, relPath string, info fs.FileIn
 	l.Debugln(w, "checking:", f)
 
 	if hasCurFile {
-		if curFile.IsEquivalentOptional(f, protocol.FileInfoComparison{
+		comp := protocol.FileInfoComparison{
 			ModTimeWindow:   w.ModTimeWindow,
 			IgnorePerms:     w.IgnorePerms,
 			IgnoreBlocks:    true,
 			IgnoreFlags:     w.LocalFlags,
 			IgnoreOwnership: !w.ScanOwnership,
 			IgnoreXattrs:    !w.ScanXattrs,
-		}) {
+		}
+		if curFile.IsEquivalentOptional(f, comp) {
 			l.Debugln(w, "unchanged:", curFile)
 			return nil
 		}
+		if !(comp.IgnoreOwnership && comp.IgnoreXattrs) {
+			comp.IgnoreInodeTime = true
+			if curFile.IsEquivalentOptional(f, comp) {
+				f.Version = curFile.Version
+				l.Debugln(w, "only inode time changed, updating without bumping version:", f)
+				select {
+				case finishedChan <- ScanResult{File: f}:
```

Comment:
Does this work? I'm sceptical -- we're sending the FileInfo onwards as if it were finished, but it doesn't have any blocks so either this depends on the change being ignored at the next level and not inserted into the database, or we lose all the block info here?

Reply:
Clearly it wouldn't have worked - would need to send the current  file with updated inode time. And I originally though this has merit on it's own on systems with xattrs or ownerships enabled, to avoid unnecessary vector bumps, however I guess it's very rare for inode time to change but none of perms, xattrs and ownership change at the same time.